### PR TITLE
✨ Added spans property to TextPrimitive

### PR DESCRIPTION
**Pull request by andretortolano**
`targeting release/1.3.0`

adds spans as a TextPrimitive property and handles it so whenever we have spans we basically ignore the content sent in "text" and override it with the combined values of all spans.

also when we have only text we map to a single span item so its straight forward way to handle all texts (as span elements).

image

### CODE REVIEW

File: appcues/src/main/java/com/appcues/data/model/ExperiencePrimitive.kt [hunk 1]
```diff
@@ -31,11 +31,17 @@ internal sealed class ExperiencePrimitive(
         override val id: UUID,
         override val style: ComponentStyle = ComponentStyle(),
         val text: String,
+        val spans: List<TextSpanPrimitive>,
```
Comment (iujames) [inline appcues/src/main/java/com/appcues/data/model/ExperiencePrimitive.kt:34]: can we get rid of `text` entirely at this point?

File: appcues/src/main/java/com/appcues/data/mapper/step/primitives/TextPrimitiveMapper.kt [hunk 2]
```diff
@@ -1,11 +1,41 @@
...
+    val processedSpans = when {
+        // when text is not null and spans is null or empty we use the raw text as a single span item
+        text != null && spans.isNullOrEmpty() -> text.toTextSpanPrimitiveList(style)
+        // else when span is not null we nao to Primitive
+        spans != null -> spans.toTextSpanPrimitive()
+        // else an empty list
+        else -> arrayListOf()
+    }
+
+    // check if this TextPrimitive is valid (contains text in form of spans)
```
Comment (iujames) [inline appcues/src/main/java/com/appcues/data/mapper/step/primitives/TextPrimitiveMapper.kt:22]: suggest adding a unit test for this text/span mapping logic

**Review by iujames — APPROVED** [review]
looks good! couple additional thoughts below

File: appcues/src/main/java/com/appcues/data/model/ExperiencePrimitive.kt (hunk 1 above)
Comment (andretortolano) [reply appcues/src/main/java/com/appcues/data/model/ExperiencePrimitive.kt:34]: it is possible, but its still useful to have either "text" the way it is, or replace  it entirely and use the contentDescription when we need the full text

Comment (iujames) [reply appcues/src/main/java/com/appcues/data/model/ExperiencePrimitive.kt:34]: is there anywhere that we still use this `text` value after mapping to spans?

**Review by iujames — APPROVED** [review]
looks great!

**Pull request merged**